app/controllers/settings/people_controller.rb

```diff
@@ -0,0 +1,19 @@
+class Settings::PeopleController < ApplicationController
+
+  def edit
+  end
+
+  def update
+    if Current.person.update(person_params)
+      redirect_to edit_settings_person_path, notice: "Person was successfully updated.", status: :see_other
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def person_params
+    params.require(:person).permit(:email, personable_attributes: [:id, :first_name, :last_name, :password, :openai_key])
```

Comment:
It took me awhile to make it work so that Person.update() would update the associated User. It turns out that passing in the :id field on User was the key to make it work. I think this opens up the possibility of submitting bad data to this endpoint to try and modify a user who is not you, so I added checks deep in the Person model to ensure that updates always use the proper id